resources/js/package.js

```diff
@@ -1,37 +1,69 @@
-document.addEventListener('turbo:load', function () {
-    ['shipping_address', 'billing_address'].forEach((type) => {
-        window.app.$watch('checkout.'+type+'.postcode', function () {
-            getAddressFromPostcodeservice(type)
-        })
+import { useThrottleFn } from "@vueuse/core"
+
+const postcodeObserver = new MutationObserver(useThrottleFn(() => {
+    let types = ['housenumber', 'postcode', 'country']
+    types.forEach((type) => {
+        document.querySelectorAll(`[id\$='_${type}']`).forEach(el => {
+            if (el.id in window.app.custom.postcode_data) {
+                return
+            }
 
-        window.app.$watch('checkout.'+type+'.street.1', function () {
-            getAddressFromPostcodeservice(type)
+            window.app.custom.postcode_data[el.id] = el.value
+            let id = el.id.replace(`_${type}`, '')
+            el.addEventListener('change', () => {
+                window.app.custom.postcode_data[el.id] = el.value
+                getAddressFromPostcodeservice(id)
+            })
         })
     })
+}, 500, true, true))
+
+document.addEventListener('turbo:load', function () {
+    window.app.custom.postcode_data = {}
+
+    postcodeObserver.observe(window.app.$el, {
+        childList: true,
+        subtree: true,
+    })
 })
 
-async function getAddressFromPostcodeservice(type) {
-    if (window.app.checkout[type].country_id != 'NL') {
+async function getAddressFromPostcodeservice(id) {
+    let get = (type) => window.app.custom.postcode_data[`${id}_${type}`]
+    let elm = (type) => document.getElementById(`${id}_${type}`)
+    let set = (type, value) => {
```

Comment:
Will you still know what these function names will mean in the future when you haven't worked on it for a while?
right here they are easy to read, but when debugging when you have a stacktrace and a function name it may be easier to have full function names